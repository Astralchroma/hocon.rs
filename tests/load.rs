use insta::assert_debug_snapshot_matches;
use test_generator;

use hocon::{self, Hocon};

test_generator::test_expand_paths! { file_load; "tests/data/*.conf" }

fn stable_readable_display(value: &Hocon) -> String {
    match value {
        Hocon::Real(v) => format!("{}", v),
        Hocon::Integer(v) => format!("{}", v),
        Hocon::String(v) => format!("\"{}\"", v),
        Hocon::Boolean(v) => format!("{}", v),
        Hocon::Array(v) => format!(
            "[{}]",
            v.iter()
                .map(|i| stable_readable_display(i))
                .collect::<Vec<_>>()
                .join(", ")
        ),
        Hocon::Hash(v) => {
            let values = v.iter().collect::<Vec<(&String, &Hocon)>>();
            let mut slice_vals = values.into_boxed_slice();
            slice_vals.sort_by(|a, b| a.0.partial_cmp(b.0).unwrap());
            format!(
                "{{{}}}",
                slice_vals
                    .iter()
                    .map(|(k, v)| format!("{}: {}", k, stable_readable_display(v)))
                    .collect::<Vec<_>>()
                    .join(", ")
            )
        }

        Hocon::BadValue => String::from("BadValue"),
    }
}

fn file_load(file_name: &str) {
    let doc = hocon::Hocon::load_from_file(file_name);

    assert!(doc.is_ok());
    assert_debug_snapshot_matches!(
        file_name.split('/').last().unwrap(),
        stable_readable_display(&doc.unwrap())
    );
}
