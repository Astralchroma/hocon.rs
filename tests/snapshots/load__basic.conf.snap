---
created: "2019-02-03T21:18:07.127928+00:00"
creator: insta@0.5.4
expression: stable_readable_display(&doc.unwrap())
source: tests/load.rs
---
"{a: 5, b: 6.7, c: [1, 2, 3], d: true, e: \"val\", f: {g: false}}"
